powerstore/datasource_snapshotrule_test.go

```diff
@@ -0,0 +1,120 @@
+package powerstore
+
+import (
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+// Test to Fetch SnapshotRule
+func TestAccSnapshotRule_FetchSnapshotRule(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: testProviderFactory,
+		Steps: []resource.TestStep{
+			{
+				Config: SnapshotRuleDataSourceparamsName,
+			},
+			{
+				Config:      SnapshotRuleDataSourceparamsNameEmpty,
+				ExpectError: regexp.MustCompile("Unable to Read PowerStore Snapshot Rules"),
+			},
+			{
+				Config: SnapshotRuleDataSourceparamsID,
+			},
+			{
+				Config:      SnapshotRuleDataSourceparamsIDEmpty,
+				ExpectError: regexp.MustCompile("Unable to Read PowerStore Snapshot Rules"),
+			},
+			{
+				Config: SnapshotRuleDataSourceparamsAll,
+			},
+			{
+				Config:      SnapshotRuleDataSourceparamsNameNegative,
+				ExpectError: regexp.MustCompile("Unable to Read PowerStore Snapshot Rules"),
+			},
+		},
+	})
+}
+
+var SnapshotRuleDataSourceparamsName = `
+provider "powerstore" {
```

Comment:
we can get rid of redundant provider block using https://github.com/dell/terraform-provider-powerstore/blob/3e474d2a8115227d4c4d46192c3597fa5ccfbae9/powerstore/provider_test.go#L33

Reply:
This is a new change so it will be added to all the resources and data sources in a separate PR to avoid any mistake